src/app/content/components/utils/attachHighlight.ts

```diff
@@ -5,12 +5,16 @@ const attachHighlight = (
   highlight: Highlight | SerializedHighlight,
   highlighter: Highlighter
 ) => {
-  highlighter.highlight(highlight);
-  const result = highlighter.getHighlight(highlight.id);
-  if (!result || !result.isAttached()) {
-    Sentry.captureException(new Error(`Highlight with id: ${highlight.id} has not been attached.`));
+  try {
+    highlighter.highlight(highlight);
+    const result = highlighter.getHighlight(highlight.id);
+    if (!result || !result.isAttached()) {
+      Sentry.captureException(new Error(`Highlight with id: ${highlight.id} has not been attached.`));
+    }
+    return result;
+  } catch (e) {
+    Sentry.captureException(e);
```

Comment:
Now we are calling Sentry with two different messages. It would be nice to call it only once, maybe using `let results: Highlight | undefined` at the beginning?
I think that `attachHighlight` should now require optional arg `errorMsg: (highlight) => string`
Default message would be `Highlight with id: ${highlight.id} has not been attached.`
This is needed to provide miningfull sentry error because search highlights have randomly generated ids

Reply:
What do you think about passing an optional callback instead? It would still be possible to throw a custom error from there and add more logic if necessary.